### PR TITLE
Implement full (stubbed out) fetch and produce request schema

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -461,6 +461,20 @@ class Log(@volatile var dir: File,
    * </ol>
    */
   private def analyzeAndValidateRecords(records: MemoryRecords): LogAppendInfo = {
+    def validateEntry(entry: LogEntry) {
```
Comment (apurvam) [inline core/src/main/scala/kafka/log/Log.scala]: What's the motivation for this change? This method seems to be called only once, and I am not sure the indirection is worth it.

**Comment by hachikuji** [comment]
@apurvam agreed. The inner def made sense in an earlier commit, but I've removed it now.

**Comment by hachikuji** [comment]
And that's a fair point about AbortedTransactions. I'll add a serde test case.
